openhands/core/config/llm_config.py

```diff
@@ -90,7 +90,11 @@ def __post_init__(self):
         """
         Post-initialization hook to assign OpenRouter-related variables to environment variables.
         This ensures that these values are accessible to litellm at runtime.
+        Also sets model-specific capabilities.
         """
+        # Set function calling support for DeepSeek models
+        if 'deepseek' in self.model.lower():
+            self.supports_function_calling = True
```

Comment:
The changes to this file are not necessary. You can revert this file, from the main branch version. Make sure to only revert this file, and commit.